vo_opengl: move eval_szexpr to user_shaders.c

haasn opened the pull request:
This moves some of the bulky user-shader specific logic into the file
dedicated to it. Rather than expose video.c state, variable lookup is
now done via a simulated closure.

The macro EVAL_SZEXPR is a drop-in replacement for the old eval_szexpr
which sets up the right enclosing state.

I'm not too sure about this commit. The code inside video.c is still ugly. Needing a macro only to fit inside 80col is sort of annoying. I'd also love to do away with the extra struct polluting the name space as well. If only I could somehow avoid the "HOOKED" specialcase?

Maybe I could have saved_tex[0] be reserved for "HOOKED" and update that one as needed instead, so none of the code would have to specialcase it? Then I could drop the struct and just pass in `struct gl_video *` directly.

wm4 commented:
> The code inside video.c is still ugly. Needing a macro only to fit inside 80col is sort of annoying. 

I wouldn't obsess over this too much. You can set the hard column limit as somewhere below 90, and in the end, this rule is supposed to make code more readable, not less.

Also, you could declare the struct just on a separate line.

What I don't like about the macro is that it's just a rather specific cosmetic indirection that's used only 3 times or so.

> Maybe I could have saved_tex[0] be reserved for "HOOKED" and update that one as needed instead, so none of the code would have to specialcase it? 

I don't know. Sounds ok to me.

haasn commented:
Rebased and removed the EVAL_SZEXP macro in favor of just breaking 80col on those few lines.

I decided against trying to specialcase HOOKED = saved_tex[0] or something weird like that just to work around using a struct. Structs are fine.

wm4 commented:
Merged.
